logstash-core/src/main/java/org/logstash/common/clients/HttpClient.java

```diff
@@ -0,0 +1,701 @@
+     * Performs an HTTP GET request
+     *
+     * @param relativePath  Relative path to resource, e.g. api/kibana/dashboards/export
+     * @param headers       Headers to include with request
+     * @return Response body
+     * @throws RequestFailedException
+     * @throws IOException
+     */
+    public String get(String relativePath, Map<String, String> headers) throws RequestFailedException, IOException {
+        String url = makeUrlFrom(relativePath);
+        HttpGet request = new HttpGet(url);
+
+        if (headers != null) {
+            headers.forEach(request::addHeader);
+        }
+
+        CloseableHttpResponse response = null;
+
+        try {
+            response = httpClient.execute(request);
+            if (response.getStatusLine().getStatusCode() >= 400) {
+                throw new RequestFailedException("HEAD", url , response.getStatusLine().getReasonPhrase());
+            }
+
+            OutputStream responseBody = new ByteArrayOutputStream();
+            response
+                .getEntity()
+                .writeTo(responseBody);
+            return ((ByteArrayOutputStream) responseBody).toString("UTF-8");
+
+        } catch (IOException e) {
+            throw new RequestFailedException("GET", url, e);
+        } finally {
+            if (response != null) {
+                response.close();
+            }
+        }
+    }
+
+    /**
+     * Performs an HTTP POST request
+     *
+     * @param relativePath  Relative path to resource, e.g. api/kibana/dashboards/import
+     * @param requestBody   Body of request
+     * @return Response body
+     * @throws RequestFailedException
+     * @throws IOException
+     */
+    public String post(String relativePath, String requestBody) throws RequestFailedException, IOException {
+        return post(relativePath, requestBody, null);
+    }
+
+    /**
+     * Performs an HTTP POST request
+     *
+     * @param relativePath  Relative path to resource, e.g. api/kibana/dashboards/import
+     * @param requestBody   Body of request
+     * @param headers       Headers to include with request
+     * @return Response body
+     * @throws RequestFailedException
+     * @throws IOException
+     */
+    public String post(String relativePath, String requestBody, Map<String, String> headers) throws RequestFailedException, IOException {
+
+        String url = makeUrlFrom(relativePath);
+
+        HttpPost request = new HttpPost(url);
+        request.setEntity(new StringEntity(requestBody));
+
+        if (headers != null) {
+            headers.forEach(request::addHeader);
+        }
+
+        CloseableHttpResponse response = null;
+
+        try {
+            response = httpClient.execute(request);
+            if (response.getStatusLine().getStatusCode() >= 400) {
+                throw new RequestFailedException("HEAD", url , response.getStatusLine().getReasonPhrase());
+            }
+
+            OutputStream responseBody = new ByteArrayOutputStream();
```

Comment:
You can just make the type here `ByteArrayOutputStream` instead of `OutputStream` and safe the cast in line 189 :)